generic.go

```diff
@@ -423,3 +423,19 @@ func GroupBy[V any, T comparable](slice []V, f func(v V) T) map[T][]V {
 
 	return result
 }
+
+// Partition returns two slices: the first containing all elements that returned
+// `true` for `f`, and the second containing all others
```

Comment:
any ideas on how to explain this better?

Reply:
Partition returns two slices: the first containing all elements that returned true through the given callback clause and the second one containing the rest of the values

any thoughts @thefuga?